### ISSUE TITLE
Snapshot doesn't get saved to form submission in remote environments

### ISSUE THREAD

**Issue by chrismlusk** [verbb/formie#1379]
### Describe the bug

I'm using the `snapshot` functionality to store arbitrary data that I am then accessing in a custom integration (see #1364).

From my template:
```twig
{% do form.setSnapshotData('extraData', {
    siteLanguage: currentSite.language,
    siteName: currentSite.name,
    submissionPath: craft.app.request.pathInfo,
}) %}

{{ craft.formie.renderForm(form) }}
```

From my integration:
```php    
$extraData = $submission->form->getSnapshotData('extraData') ?? [];
$payload = array_merge($payload, $extraData);

// send the payload ...
```

In local dev, everything works as expected. The service receiving the payload includes the extra data from the snapshot. However, when submitting the same form on staging and production, that extra data is missing.

I checked the `formie_submission` table in the database, and in my local db, the Snapshot column has the extra data. The staging and production databases have an empty array in that column, though.

### Steps to reproduce

See above.

### Form settings

- Multi-page form: No
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: No

### Craft CMS version

4.4.6.1

### Plugin version

2.0.26

### Multi-site?

Yes

### Additional context

Since this isn't working on remote environments, it's worth mentioning this site is hosted on Servd.

**Comment by engram-design** [comment]
How strange, that seems quite strange behaviour even on a different environment. It uses the same session storage mechanism we use to keep track of incomplete submissions for a multi-page form. I'll have to spin up a Servd site.

**Comment by chrismlusk** [comment]
Hi @engram-design. Curious if you've had a chance to investigate this?

**Comment by engram-design** [comment]
I haven't had the chance to test this out on Servd yet, sorry. Are you using anything like custom session storage (like Redis) or anything on Servd? That's probably the only thing I can think of that might cause a difference in behaviour.

In addition, are you running the queue in production differently to local development? Integrations by default are run via Craft's queue, and maybe that's also a factor here (run on the command line).

**Comment by chrismlusk** [comment]
Aha. Your queue question might have led me to the issue.

Servd offers — and this project uses — a dedicated queue runner that runs isolated from user requests, so jobs aren't being run as web requests - https://servd.host/docs/dedicated-queue-runner#troubleshooting

**Comment by engram-design** [comment]
That gives me something to test as well. But I _still_ don't think it should be an issue, because the snapshot data would've already been stored in the database against the submission, so it no longer relies on session data.

Out of interest though, i've noticed you're fetching the snapshot data from the form, which is uncesssary. You can just fetch it from the submission (where it's stored).

```diff
- $extraData = $submission->form->getSnapshotData('extraData') ?? [];
+ $extraData = $submission->snapshot['extraData'] ?? [];
```

And I'm wondering if that's the issue, because the Form's snapshot relies on session data, to resolve it to the submission.

**Comment by chrismlusk** [comment]
That's fair. Although since the `formie_submission` table in my production/staging DBs had empty arrays in the Snapshot column, it still feels like another issue?

I've since taken a different approach to pass this extra data around, so I can't test it at the moment. Also, I'm heading out for a couple weeks off, so it'll be a bit before I can follow up with here. FYI!

**Comment by engram-design** [comment]
Right, so if the column is empty, that means that it's failing to populate that from the form, which is indeed a sessions issue. I'll be sure to spin up a Servd site to see what's going on.

**Comment by engram-design** [comment]
Updated in [4.0.0-beta.10](https://github.com/verbb/formie/releases/tag/4.0.0-beta.10).
